src/main/java/org/zalando/riptide/Rest.java

```diff
@@ -58,15 +77,26 @@ protected UriTemplateHandler getUriTemplateHandler() {
      * response thrown by the {@link OAuth2CompatibilityResponseErrorHandler} and continues with normal dispatching.
      * </p>
      */
-    private <T> ClientHttpResponse execute(final HttpMethod method, final URI url, final Callback<T> callback) {
+    private <T> ClientHttpResponse executeRequest(final HttpMethod method, final URI url, HttpEntity<T> entity) {
         try {
-            return template.execute(url, method, callback, BufferingClientHttpResponse::buffer);
-        } catch (final AlreadyConsumedResponseException e) {
-            return e.getResponse();
+            final ClientHttpRequest request = clientHttpRequestFactory.createRequest(url, method);
+            writeRequestEntity(entity, request, converters);
+            return request.execute();
+        } catch (IOException ex) {
+            final String message = String.format("I/O error on %s request for \"%s\": %s", method.name(), url, ex.getMessage());
+            throw new ResourceAccessException(message, ex);
         }
     }
 
+    public static Rest create(final ClientHttpRequestFactory clientHttpRequestFactory, final List<HttpMessageConverter<?>> converters, final UriTemplateHandler uriTemplateHandler) {
+        return new Rest(clientHttpRequestFactory, converters, uriTemplateHandler);
+    }
+
+    public static Rest create(final ClientHttpRequestFactory clientHttpRequestFactory, final List<HttpMessageConverter<?>> converters) {
+        return create(clientHttpRequestFactory, converters, new DefaultUriTemplateHandler());
+    }
+
     public static Rest create(final RestTemplate template) {
```

Comment:
Maybe best to pick something else than `create` (clarifies it also fur users of the legacy riptide).